cmake/modules/CheckCompiler.cmake

```diff
@@ -8,6 +8,10 @@
 #  CheckCompiler.cmake
 #---------------------------------------------------------------------------------------------------
 
+if(NOT CMAKE_CXX_COMPILER_ID MATCHES "(Apple|)Clang|GNU|Intel|MSVC")
+  message(WARNING "Unsupported compiler: ${CMAKE_CXX_COMPILER_ID}.")
```

Comment:
```suggestion
  message(WARNING "Untested compiler: ${CMAKE_CXX_COMPILER_ID}. Setup may be incomplete.")
```

Reply:
Yeah, downgrading the message to a warning had this intent. But unsupported is still probably more accurate, because it literally means that we don't have code in the build system to support the compiler, as is the case for GCC, ICC, Clang, and MSVC, so it's not just that the other compilers are untested.